cvat-ui/src/utils/tasks-dto.ts

```diff
@@ -0,0 +1,162 @@
+export enum FileSource {
+  Local = 1,
+  Remote = 2,
+  Share = 3,
+}
+
+export function fileModel(parentNode: any, files: any) {
+  return files.map(
+    (file: any) => {
+      return {
+        id: parentNode.props ? `${parentNode.props.title}/${file.name}` : file.name,
+        isLeaf: file.type !== 'DIR',
+        name: file.name,
+      };
+    }
+  );
+}
+
+export function taskDTO(values: any) {
+  const newTaskDTO = {
+    name: values.name,
+    labels: deserializeLabels(values.labels),
+    image_quality: values.imageQuality,
+    z_order: values.zOrder,
+    bug_tracker: values.bugTracker,
+    segment_size: values.segmentSize,
+    overlap: values.overlapSize,
+    frame_filter: values.frameFilter,
+    start_frame: values.startFrame,
+    stop_frame: values.stopFrame,
+  };
+
+  const newTask = new (window as any).cvat.classes.Task(newTaskDTO);
+
+  if (values.source === FileSource.Local) {
+    newTask.clientFiles = values.localUpload.fileList.map((file: any) => file.response);
+  } else if (values.source === FileSource.Remote) {
+    newTask.remoteFiles = values.remoteURL
+      .split(/\r?\n/)
+      .map((url: string) => url.trim())
+      .filter((url: string) => url.length > 0);
+  } else if (values.source === FileSource.Share) {
+    newTask.serverFiles = values.sharedFiles;
+  }
+
+  return newTask;
+}
+
+export function validateLabels(rule: any, value: string, callback: Function) {
+  if (value) {
+    try {
+      deserializeLabels(value);
+    } catch (error) {
+      callback(error.message);
+    }
+  }
+
+  callback();
+}
+
+export function convertStringToNumber(event: React.FormEvent<HTMLInputElement>) {
+  const convertedValue = Number(event.currentTarget.value);
+
+  if (isNaN(convertedValue)) {
+    return 0;
```

Comment:
It can return 0 for segment_size and image_quality if I enter a string. And validator says nothing, right?

Reply:
@bsekachev Is this incorrect?

Reply:
@zankevich 
0 is out of range for these inputs

Reply:
UPD: I noticed, image quality can be 0